traffic_portal/test/end_to_end/users/users-spec.js

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var pd = require('./pageData.js');
+var cfunc = require('../common/commonFunctions.js');
+
+describe('Traffic Portal Users Test Suite', () => {
+    const pageData = new pd();
+    const commonFunctions = new cfunc();
+    const newUser = function() {
+        return {
+            name: 'User Name',
+            username: 'username-' + commonFunctions.shuffle('abcdefghijklmonpqrstuvwxyz0123456789'),
+            email: 'user-' + commonFunctions.shuffle('abcdefghijklmonpqrstuvwxyz0123456789') + '@example.com',
+            password: commonFunctions.shuffle('abcdefghijklmonpqrstuvwxyz0123456789')
+        };
+    };
+    const myNewUser = newUser();
+    const myOtherNewUser = newUser();
+
+    it('should register a new user', async () => {
+        await browser.setLocation('users');
+        await pageData.registerUserButton.click();
+        await pageData.email.sendKeys(myNewUser.email);
+        commonFunctions.selectDropdownbyNum(pageData.role, 3);
```

Comment:
can you select by tenant name / role name instead? by id always makes me nervous because if for some reason the id is different (or not there), the test will break. by name makes me think the tests will be less brittle...and brittle tests are never good.

Reply:
agree with jeremy. selecting by index isn't recommended unless it is a requirement to have specific label tied to specific index.